### PR TITLE
Add basic unit tests

### PR DESCRIPTION
**Pull request by picklelo**
Added unit tests for all the base classes and made sure they pass.

### CODE REVIEW

File: pynecone/state.py [hunk 1]
```diff
@@ -368,13 +368,17 @@ def get_delta(self) -> Delta:
...
             The delta for the state.
         """
+        delta = {}
+
         # Return the dirty vars, as well as all computed vars.
-        delta = {
-            self.get_full_name(): {
-                prop: getattr(self, prop)
-                for prop in self.dirty_vars | set(self.computed_vars.keys())
+        if len(self.dirty_vars) > 0:
```
Comment (picklelo) [inline pynecone/state.py:374]: I just realized this may break the clock app - maybe I'll get rid of this `if` statement.

File: tests/test_app.py [hunk 2]
```diff
@@ -36,25 +39,32 @@ def about():
...
     Args:
         app: The app to test.
     """
+    assert app.state() == DefaultState()
     assert app.middleware == [HydrateMiddleware()]
+    assert app.style == Style()
 
 
-def test_add_page_default_route(app: App, index_page, about_page) -> None:
+def test_add_page_default_route(app: App, index_page, about_page):
```
Comment (Alek99) [inline tests/test_app.py:67]: Maybe for later but a nested route test here.

File: tests/test_utils.py [hunk 3]
```diff
@@ -169,3 +169,13 @@ def test_format_cond(condition: str, true_value: str, false_value: str, expected
         expected: The expected output string.
     """
     assert utils.format_cond(condition, true_value, false_value) == expected
+
```
Comment (Alek99) [inline tests/test_utils.py:172]: I can't seem to put this comment where it belongs on line 9. I think we should add a test for snake case with a _ coming first for example  (_hover)

File: pynecone/var.py [hunk 4]
```diff
@@ -153,14 +153,15 @@ def __getitem__(self, i: Any) -> Var:
...
-        elif utils.is_dataframe(self.type_):
+        elif utils._issubclass(self.type_, Dict) or utils.is_dataframe(self.type_):
             if isinstance(i, str):
                 i = utils.wrap(i, '"')
             if isinstance(self.type_, _GenericAlias):
                 type_ = utils.get_args(self.type_)[1]
             else:
                 type_ = Any
         else:
+            breakpoint()
```
[the remark this answers is not in the document]
Comment (picklelo) [reply pynecone/var.py]: Good catch, removed.

File: pynecone/state.py (hunk 1 above)
Comment (picklelo) [reply pynecone/state.py]: I removed this to always include computed vars.

File: tests/components/test_component.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    """
+    style = {
+        component1: Style({"color": "white"}),
+        component2: Style({"color": "black"}),
+    }
+    c1 = component1().add_style(style)  # type: ignore
+    c2 = component2().add_style(style)  # type: ignore
+    assert c1.style["color"] == "white"
+    assert c2.style["color"] == "black"
+
```
[the remark this answers is not in the document]
Comment (picklelo) [reply tests/components/test_component.py:114]: Added a test for this.

File: tests/test_app.py (hunk 2 above)
Comment (picklelo) [reply tests/test_app.py:67]: Added a test for nested routes below.

File: tests/test_utils.py (hunk 3 above)
Comment (picklelo) [reply tests/test_utils.py:174]: Good idea, added a test for this

**Review by Alek99 — APPROVED** [review]

**Pull request merged**